### PR TITLE
Support async generator functions in `Response` and `Request` for bodies

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

This makes the following code work:
```js
await new Response(
  async function* stream() {
    yield "hey";
    await Bun.sleep(42);
    yield Buffer.alloc(100);
  }
).arrayBuffer();
```

```js
await new Response({
  async *[Symbol.asyncIterator]() {
    yield "hey";
    await Bun.sleep(42);
    yield Buffer.alloc(100);
  },
}).arrayBuffer();
```

### How did you verify your code works?

There are tests. The ones for .json() fail. Need to fix that before we can merge

### CODE REVIEW

**Review by paperclover — COMMENTED** [review]
did not read the test. looks good. i'm gonna give this a go on astro before marking this as approved

File: test/js/bun/http/async-iterator-stream.test.ts [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  });
+
+  test("async generator function throws an error but continues to send the headers", async () => {
+    const server = Bun.serve({
+      port: 0,
+
+      async fetch(req) {
+        return new Response(
+          async function* () {
+            throw new Error("Oops");
```
Comment (Electroid) [inline test/js/bun/http/async-iterator-stream.test.ts:38]: Can we either move this test to a subprocess or add a special property that hides the errors from being printed? This makes our GitHub CI spammy with errors

File: test/js/bun/http/async-iterator-stream.test.ts [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+          {
+            headers: {
+              "X-Hey": "123",
+            },
+          },
+        );
+      },
+    });
+
+    const res = await fetch(`${server.url}/`);
```
Comment (Electroid) [inline test/js/bun/http/async-iterator-stream.test.ts]:
```suggestion
    const res = await fetch(server.url);
```

File: test/js/bun/http/async-iterator-stream.test.ts (hunk 1 above)
Comment (Jarred-Sumner) [reply test/js/bun/http/async-iterator-stream.test.ts:38]: That should be fixed in the GitHub actions printer IMO

**Review by paperclover — APPROVED** [review]
nice

File: test/js/bun/http/async-iterator-stream.test.ts (hunk 1 above)
Comment (jdalton) [reply test/js/bun/http/async-iterator-stream.test.ts:38]: Related to https://github.com/oven-sh/bun/pull/8949

**Pull request merged**